### PR TITLE
[nas] fix issue introduced by the trial recovery feature

### PR DESCRIPTION
**Pull request by QuanluZhang**
fix bugs in #4931

### CODE REVIEW

**Review by liuzhe-lz — APPROVED** [review]

File: nni/nas/execution/common/integration.py [hunk 1]
```diff
@@ -239,4 +256,6 @@ def handle_import_data(self, data):
         pass
 
     def handle_add_customized_trial(self, data):
-        pass
+        previous_max_param_id = self.recover_parameter_id(data)
+        self.parameters_count = previous_max_param_id
+        self._advisor_initialized = True
```
Comment (ultmaster) [inline nni/nas/execution/common/integration.py:261]: Is it possible that `handld_add_customized_trial` is never called and `_advisor_initialized` is never set to true?

File: nni/nas/execution/common/integration_api.py [hunk 2]
```diff
@@ -41,7 +45,11 @@ def send_trial(parameters: dict, placement_constraint=None) -> int:
     Send a new trial. Executed on tuner end.
     Return a ID that is the unique identifier for this trial.
     """
-    return get_advisor().send_trial(parameters, placement_constraint)
+    advisor = get_advisor()
+    while not advisor.initialized:
```
Comment (ultmaster) [inline nni/nas/execution/common/integration_api.py:49]: Suggest putting this into `RetiariiAdvisor.send_trial`.

File: nni/nas/execution/common/integration.py (hunk 1 above)
Comment (QuanluZhang) [reply nni/nas/execution/common/integration.py]: good point! `handle_add_customized_trial` is called when experiment is resumed, but will not be called when experiment is created, this is a bug introduced by me...
I moved this flag to `handle_request_trial_jobs`, which means if trial is not requested, `send_trial` will be blocked. And "request trial" will always be sent by nnimanager

File: nni/nas/execution/common/integration_api.py (hunk 2 above)
Comment (QuanluZhang) [reply nni/nas/execution/common/integration_api.py]: updated

**Review by ultmaster — APPROVED** [review]

**Pull request merged**